manifests/instance.pp

```diff
@@ -46,6 +47,35 @@
     }
   }
 
+  if $manage_service {
+    if $instance_user_home == undef{
+      fail('Please set instance_user_home in order to manage the db2 service instance')
+    }
+
+    $instance_service = "db2_${name}.service"
+
+    file{"/etc/systemd/system/${instance_service}":
+      ensure  => present,
+      content => template('db2/db2_instance.service.erb'),
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0644',
+      notify  => Exec['db2_systemd_daemon_reload'],
+    }
+
+    service{$instance_service:
+      ensure    => 'running',
+      enable    => true,
+      subscribe => [
+        Exec['db2_systemd_daemon_reload'],
+        Db2_instance[$instance_user],
```

Comment:
This suggests that the service needs to be restarted after adding an instance, dcs, catalog or database.... Is this really the case?

Reply:
I thought so, but now that you ask I will check this with the database specialists so we get the behaviour right.